pallets/xcm-executor-utils/src/lib.rs

```diff
@@ -0,0 +1,263 @@
+// Copyright (C) Moondance Labs Ltd.
+// This file is part of Tanssi.
+
+// Tanssi is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Tanssi is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Tanssi.  If not, see <http://www.gnu.org/licenses/>.
+
+//! # Authorities Noting Pallet
+//!
+//! This pallet notes the authorities assigned to this container-chain in an orchestrator chain
+//!
+//! First the pallet receives a storage proof of the header of the orchestrator chain
+//! Once the storage proof is verified against the relay, the storage root of the orchestrator
+//! chain is retrieved from the header
+//!  
+//! A second storage proof is verified against the storage root of the orchestrator chain. From
+//! this the collator-assignation is read, and the authorities assigned to these container-chain
+//! are retrieved and stored
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[cfg(test)]
+mod mock;
+
+#[cfg(test)]
+mod tests;
+
+#[cfg(any(test, feature = "runtime-benchmarks"))]
+mod benchmarks;
+
+pub mod filters;
+
+pub use pallet::*;
+
+use {
+    frame_support::pallet_prelude::*,
+    frame_system::pallet_prelude::*,
+    serde::{Deserialize, Serialize},
+    staging_xcm::latest::{AssetId, MultiLocation},
+};
+
+#[frame_support::pallet]
+pub mod pallet {
+    use super::*;
+    use sp_runtime::BoundedVec;
+    use sp_std::vec::Vec;
+
+    // Default filtering policies for incoming assets
+    #[derive(
+        PartialEq,
+        Eq,
+        Clone,
+        Encode,
+        Decode,
+        RuntimeDebug,
+        TypeInfo,
+        MaxEncodedLen,
+        Serialize,
+        Deserialize,
+    )]
+    pub enum DefaultFilterPolicy {
+        // Allow all incoming assets
+        All,
+        // Only allow assets native of the origin
+        AllNative,
+        // Do not allow any assets
+        Never,
+    }
+
+    #[derive(
+        DebugNoBound,
+        CloneNoBound,
+        EqNoBound,
+        PartialEqNoBound,
+        Encode,
+        Decode,
+        TypeInfo,
+        MaxEncodedLen,
+        Serialize,
+        Deserialize,
+    )]
+    #[scale_info(skip_type_params(MaxAssets))]
+    #[serde(bound = "")]
+    pub enum FilterPolicy<MaxAssets: Get<u32>> {
+        DefaultFilterPolicy(DefaultFilterPolicy),
+        AllowedAssets(BoundedVec<AssetId, MaxAssets>),
```

Comment:
It was cool to get the `BoundedVec` to work with `T::FilterPolicyMaxAssets` (thanks @tmpolaczyk @girazoki!), but it is kind of pain to have to handle the generic type everywhere `FilterPolicy` is used, so thinking of just dropping it and use a const.
What do you think?

Reply:
I would leave it if this doesn't affect users (in other words, we will be the only ones that need to type `FilterPolicy<T::FilterPolicyMaxAssets>`)